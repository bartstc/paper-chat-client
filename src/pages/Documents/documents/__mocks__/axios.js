module.exports = {
  get: () => {
    return Promise.resolve({
      data: [
        {
          id: 1,
          category: 'WORK',
          title: 'test'
        },
        {
          id: 2,
          category: 'HOBBY',
          title: 'test2'
        }
      ]
    });
  }
};
