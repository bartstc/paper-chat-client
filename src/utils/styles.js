import { createGlobalStyle } from 'styled-components';

export const color = {
  white: `#FFF`,
  black: `#151615`,
  grey: '#565656',
  accent: `#0075C4`,
};

export const fontFamily = {
  primary: `'Montserrat', sans-serif`,
};

export const fontWeight = {
  light: `300`,
  bold: `700`,
};

const size = {
  mobileS: '320px',
  mobileM: '375px',
  mobileL: '425px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1440px',
  laptopXL: '1644px',
  desktop: '2560px',
};

export const device = {
  mobileS: `(min-width: ${size.mobileS})`,
  mobileM: `(min-width: ${size.mobileM})`,
  mobileL: `(min-width: ${size.mobileL})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  laptopL: `(min-width: ${size.laptopL})`,
  laptopXL: `(min-width: ${size.laptopXL})`,
  desktop: `(min-width: ${size.desktop})`,
  desktopL: `(min-width: ${size.desktop})`,
};

export const GlobalStyle = createGlobalStyle`
  body {
    color: ${color.black};
    font-family: ${fontFamily.primary};
    font-weight: ${fontWeight.light};
    overflow-x: hidden;
  }
`;

export const accent = `
  color: ${color.accent};
`;
