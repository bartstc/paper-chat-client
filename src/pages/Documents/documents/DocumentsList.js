import React, { useEffect } from 'react';
import axios from 'axios';
import { DocsList, Warning } from './DocumentsList.styles';
import {
  useDocumentsState,
  useDocumentsDispatch
} from '../../../context/documentsContext';
import SpinnerSm from '../../../components/Spinner/SpinnerSm';
import DocumentItem from './documentsList/DocumentItem';

const DocumentsList = () => {
  const dispatch = useDocumentsDispatch();
  const { documents, loading } = useDocumentsState();

  useEffect(() => {
    fetchDocuments();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const fetchDocuments = async () => {
    try {
      const { data } = await axios.get('/documents');
      dispatch({ type: 'GET_DOCUMENTS', payload: data });
    } catch (err) {
      console.log(err.response.data);
    }
  };

  if (!documents || loading) return <SpinnerSm />;

  return (
    <>
      {documents && documents.length === 0 ? (
        <Warning>No documents added yet.</Warning>
      ) : (
        <DocsList>
          {documents.map(doc => (
            <DocumentItem
              key={doc.id}
              {...doc}
              fetchDocuments={fetchDocuments}
            />
          ))}
        </DocsList>
      )}
    </>
  );
};

export default DocumentsList;
