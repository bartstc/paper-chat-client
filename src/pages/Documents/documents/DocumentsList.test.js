import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';
import DocumentsList from './DocumentsList';
import {
  DocumentsStateContext,
  DocumentsDispatchContext
} from '../../../context/documentsContext';
import { Warning } from './DocumentsList.styles';
import axios from 'axios';

jest.mock('axios');

// https://github.com/f/react-hooks-todo-app

let state = {
  documents: [
    {
      id: 1,
      category: 'WORK',
      title: 'test'
    },
    {
      id: 2,
      category: 'HOBBY',
      title: 'test2'
    }
  ],
  loading: false
};

let dispatch = jest.fn();

afterEach(() => {
  jest.clearAllMocks();
});

let DocumentsProvider = ({ children }) => {
  return (
    <DocumentsStateContext.Provider value={state}>
      <DocumentsDispatchContext.Provider value={dispatch}>
        {children}
      </DocumentsDispatchContext.Provider>
    </DocumentsStateContext.Provider>
  );
};

describe('<DocumentsList />', () => {
  let wrapper = mount(
    <Router>
      <DocumentsProvider>
        <DocumentsList />
      </DocumentsProvider>
    </Router>
  );

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders list of documents when it`s not empty', () => {
    const list = wrapper.find('ul');

    expect(list.length).toEqual(1);
    expect(list.find('li').length).toEqual(2);
  });

  describe('when rendered', () => {
    it('should fetch a list of documents', () => {
      const getSpy = jest.spyOn(axios, 'get');

      wrapper = mount(
        <Router>
          <DocumentsProvider>
            <DocumentsList />
          </DocumentsProvider>
        </Router>
      );

      expect(getSpy).toBeCalled();
    });

    it('renders spinner when list of documents is `null` and loading is `true`', () => {
      state = {
        documents: null,
        loading: true
      };

      DocumentsProvider = ({ children }) => {
        return (
          <DocumentsStateContext.Provider value={state}>
            <DocumentsDispatchContext.Provider value={dispatch}>
              {children}
            </DocumentsDispatchContext.Provider>
          </DocumentsStateContext.Provider>
        );
      };

      wrapper = mount(
        <Router>
          <DocumentsProvider>
            <DocumentsList />
          </DocumentsProvider>
        </Router>
      );

      expect(wrapper.find('SpinnerSm').length).toEqual(1);
    });
  });

  it('renders warning message when list of documents is empty', () => {
    state = {
      documents: [],
      loading: false
    };

    DocumentsProvider = ({ children }) => {
      return (
        <DocumentsStateContext.Provider value={state}>
          <DocumentsDispatchContext.Provider value={dispatch}>
            {children}
          </DocumentsDispatchContext.Provider>
        </DocumentsStateContext.Provider>
      );
    };

    wrapper = mount(
      <Router>
        <DocumentsProvider>
          <DocumentsList />
        </DocumentsProvider>
      </Router>
    );

    expect(wrapper.find(Warning).length).toEqual(1);
  });
});
