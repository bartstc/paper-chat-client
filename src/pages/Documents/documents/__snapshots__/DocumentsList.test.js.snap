// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentsList /> renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DocumentsProvider>
      <DocumentsList>
        <styled.ul>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "uTxCW",
                  "rules": Array [],
                },
                "displayName": "styled.ul",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <ul
              className="sc-bwzfXH uTxCW"
            >
              <DocumentItem
                category="WORK"
                fetchDocuments={[Function]}
                id={1}
                key="1"
                title="test"
              >
                <styled.li
                  key="1"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": true,
                          "lastClassName": "gBJDIy",
                          "rules": Array [
                            "
  position: relative;
  margin-bottom: 0.8em;
  height: 40px;
  display: flex;
  align-items: center;
",
                          ],
                        },
                        "displayName": "styled.li",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "li",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <li
                      className="sc-htpNat gBJDIy"
                    >
                      <styled.span
                        color="#a30402"
                      >
                        <StyledComponent
                          color="#a30402"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": false,
                                "lastClassName": "koEcDi",
                                "rules": Array [
                                  "
  display: block;
  height: 21px;
  width: 45px;
  background: ",
                                  [Function],
                                  ";
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="sc-bxivhb dGCwnE"
                            color="#a30402"
                          />
                        </StyledComponent>
                      </styled.span>
                      <Styled(Link)
                        to="/editor/1"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-ifAKCX",
                                "isStatic": true,
                                "lastClassName": "sLouT",
                                "rules": Array [
                                  "
  padding: 0 1.4em 0 0.8em;
",
                                ],
                              },
                              "displayName": "Styled(Link)",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-ifAKCX",
                              "target": [Function],
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          to="/editor/1"
                        >
                          <Link
                            className="sc-ifAKCX sLouT"
                            to="/editor/1"
                          >
                            <a
                              className="sc-ifAKCX sLouT"
                              href="/editor/1"
                              onClick={[Function]}
                            >
                              test
                            </a>
                          </Link>
                        </StyledComponent>
                      </Styled(Link)>
                      <styled.button
                        onClick={[Function]}
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-EHOje",
                                "isStatic": true,
                                "lastClassName": "fFlAGJ",
                                "rules": Array [
                                  "
  width: 20px;
  height: 20px;
  position: absolute;
  top: 10px;
  right: 0;
  border: none;
  color: ",
                                  "#FFF",
                                  ";
  font-weight: ",
                                  "700",
                                  ";
  font-size: 0.85rem;
  background: #a10000;
  cursor: pointer;
",
                                ],
                              },
                              "displayName": "styled.button",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-EHOje",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          onClick={[Function]}
                        >
                          <button
                            className="sc-EHOje fFlAGJ"
                            onClick={[Function]}
                          >
                            X
                          </button>
                        </StyledComponent>
                      </styled.button>
                    </li>
                  </StyledComponent>
                </styled.li>
              </DocumentItem>
              <DocumentItem
                category="HOBBY"
                fetchDocuments={[Function]}
                id={2}
                key="2"
                title="test2"
              >
                <styled.li
                  key="2"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": true,
                          "lastClassName": "gBJDIy",
                          "rules": Array [
                            "
  position: relative;
  margin-bottom: 0.8em;
  height: 40px;
  display: flex;
  align-items: center;
",
                          ],
                        },
                        "displayName": "styled.li",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "li",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <li
                      className="sc-htpNat gBJDIy"
                    >
                      <styled.span
                        color="#47990c"
                      >
                        <StyledComponent
                          color="#47990c"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": false,
                                "lastClassName": "koEcDi",
                                "rules": Array [
                                  "
  display: block;
  height: 21px;
  width: 45px;
  background: ",
                                  [Function],
                                  ";
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="sc-bxivhb koEcDi"
                            color="#47990c"
                          />
                        </StyledComponent>
                      </styled.span>
                      <Styled(Link)
                        to="/editor/2"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-ifAKCX",
                                "isStatic": true,
                                "lastClassName": "sLouT",
                                "rules": Array [
                                  "
  padding: 0 1.4em 0 0.8em;
",
                                ],
                              },
                              "displayName": "Styled(Link)",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-ifAKCX",
                              "target": [Function],
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          to="/editor/2"
                        >
                          <Link
                            className="sc-ifAKCX sLouT"
                            to="/editor/2"
                          >
                            <a
                              className="sc-ifAKCX sLouT"
                              href="/editor/2"
                              onClick={[Function]}
                            >
                              test2
                            </a>
                          </Link>
                        </StyledComponent>
                      </Styled(Link)>
                      <styled.button
                        onClick={[Function]}
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-EHOje",
                                "isStatic": true,
                                "lastClassName": "fFlAGJ",
                                "rules": Array [
                                  "
  width: 20px;
  height: 20px;
  position: absolute;
  top: 10px;
  right: 0;
  border: none;
  color: ",
                                  "#FFF",
                                  ";
  font-weight: ",
                                  "700",
                                  ";
  font-size: 0.85rem;
  background: #a10000;
  cursor: pointer;
",
                                ],
                              },
                              "displayName": "styled.button",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-EHOje",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          onClick={[Function]}
                        >
                          <button
                            className="sc-EHOje fFlAGJ"
                            onClick={[Function]}
                          >
                            X
                          </button>
                        </StyledComponent>
                      </styled.button>
                    </li>
                  </StyledComponent>
                </styled.li>
              </DocumentItem>
            </ul>
          </StyledComponent>
        </styled.ul>
      </DocumentsList>
    </DocumentsProvider>
  </Router>
</BrowserRouter>
`;
